backend/src/routes/Contact.ts

```diff
@@ -7,43 +8,87 @@ import {
 import { requireSelf } from './middlewares';
 
 import type { Contact } from '../services';
+import {
+  stringToNumberTransformer,
+  validateRequest,
+} from './helpers/validation';
+
+const GET_SCHEMA = z.object({
+  params: z.object({
+    userId: stringToNumberTransformer,
+  }),
+});
+
+const POST_SCHEMA = z.object({
+  params: z.object({
+    userId: stringToNumberTransformer,
+  }),
+  body: z.object({
+    name: z.string(),
+    phoneNumber: z.string(),
+    avatar: z.string(),
+  }),
+});
+
+const PUT_SCHEMA = z.object({
+  params: z.object({
+    userId: stringToNumberTransformer,
+    contactId: stringToNumberTransformer,
+  }),
+  body: z.object({
+    name: z.string().optional(),
+    phoneNumber: z.string().optional(),
+    avatar: z.string().optional(),
```

Comment:
Let's reduce the duplication by using [partials](https://github.com/vriad/zod#partials)

Reply:
Cool added